StorageTests/TestSQLiteBookmarks.swift

```diff
@@ -77,6 +79,188 @@ class TestSQLiteBookmarks: XCTestCase {
+            "bookmark_good_additional", "http://example.org/6", "Bookmark additional 1", BookmarkRoots.MobileFolderGUID,
+            "bookmark_good_additional_over_limit", "http://example.org/7", "Bookmark additional 2", BookmarkRoots.MobileFolderGUID,
+        ]
+
+        let structureQuery =
+        "INSERT INTO \(TableBookmarksLocalStructure) (parent, child, idx) VALUES " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?) "
+
+        let structureArgs: Args = [
+            BookmarkRoots.MobileFolderGUID, "folder123", 0,
+            "folder123", "bookmark123", 0,
+            BookmarkRoots.ToolbarFolderGUID, "bookmark_other_folder", 0,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_nonsynced", 1,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_synced", 2,
+            BookmarkRoots.MobileFolderGUID, "bookmark_in_buffer", 3,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_additional", 4,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_additional_over_limit", 5,
+        ]
+
+        let bufferQuery =
+        "INSERT INTO \(TableBookmarksBuffer) (guid, type, bmkUri, title, parentid, parentName, server_modified) VALUES " +
+        "(?, \(BookmarkNodeType.bookmark.rawValue), ?, ?, ?, '', ?) "
+
+        let bufferArgs: Args = [
+            "bookmark_in_buffer", "http://example.org/5", "Bookmark in buffer(nok)", BookmarkRoots.MobileFolderGUID, Date.now()
+        ]
+
+        let bufferStructureQuery =
+        "INSERT INTO \(TableBookmarksBufferStructure) (parent, child, idx) VALUES " +
+        "(?, ?, ?)"
+
+        let bufferStructureArgs: Args = [
+            "bookmark_in_buffer", "bookmark_in_buffer", 0, // It's its own parent because we are lazy.
+        ]
+
+        db.run([
+            (sql: localQuery, args: localArgs),
+            (sql: structureQuery, args: structureArgs),
+            (sql: bufferQuery, args: bufferArgs),
+            (sql: bufferStructureQuery, args: bufferStructureArgs),
+        ]).succeeded()
+
+        let expectation = self.expectation(description: #function)
+        bookmarks.getLocalBookmarksAdditions(limit: 2) >>== { results in
+            XCTAssertEqual(results.count, 2)
+            XCTAssertEqual(results.map { $0.guid }, ["bookmark_good_nonsynced", "bookmark_good_additional"])
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 10, handler: nil)
+    }
+
+    func testApplyBufferUpdatedCompletionOp() {
+        guard let db = getBrowserDB("TSQLBtestApplyBufferUpdatedCompletionOp.db", files: self.files) else {
+            XCTFail("Unable to create browser DB.")
+            return
+        }
+
+        let localQuery =
+        "INSERT INTO \(TableBookmarksLocal) (guid, type, bmkUri, title, parentid, parentName, sync_status) " +
```

Comment:
Use `SQLiteBookmarks.addToMobileBookmarks` instead of doing this. If you really must go direct, use  `insertBookmarkInTransaction`. The point is to avoid baking duplicate logic (like `sync_status` values) into tests that might cause the tests to continue to pass when the upstream logic changes.